src/instantsearch/instantsearch.ts

```diff
@@ -179,11 +179,11 @@ export type InstantSearchConfig = {
     | boolean
     | {
         router?: {
-          onUpdate?: Function;
-          read?: Function;
-          write?: Function;
-          createURL?: Function;
-          dispose?: Function;
+          onUpdate?: (cb: (object) => void) => void;
+          read?: () => object;
+          write?: (routeState: object) => void;
+          createURL?: (routeState: object) => string;
+          dispose?: () => void;
```

Comment:
we said we'd make them all optional but I'm not so sure of it.
If I read the code correctly [the default router value is entirely overwritten by the config passed](url)
So for example if user doesn't define `onUpdate` this [line](https://github.com/algolia/instantsearch.js/blob/9632ade6434fb9257bdb23c66dde8aaa1c31c4eb/src/lib/RoutingManager.js#L52) will fail.

So making onUpdate optional is misleading

Reply:
Yep we comment at the same time see https://github.com/algolia/angular-instantsearch/pull/393#discussion_r239430417